src/seat.c

```diff
@@ -263,20 +263,25 @@ new_keyboard(struct seat *seat, struct wlr_input_device *device, bool virtual)
 	return (struct input *)keyboard;
 }
 
+static void
+map_touch_to_output(struct seat *seat, struct wlr_input_device *dev)
+{
+	struct wlr_touch *touch = wlr_touch_from_input_device(dev);
+	char *output_name = rc.touch.output_name ? rc.touch.output_name : touch->output_name;
```

Comment:
Reverse fallback here:

`char *output_name = touch.output_name ? : rc.touch->output_name;`

`touch.output_name` is set when running nested (thanks @Consolatis ) so we don't want to affect input-output mapping for that.